### ISSUE TITLE
Announcement Multi Forum Select

### ISSUE THREAD

**Issue by Ben-MyBB** [mybb/mybb#3599]
In the ACP Announcement page the forum select isn't multi-selectable.

Quote:Forums to Appear in *
The announcement will appear in the forums selected below. The announcement will appear in the selected forum and all child forums of the selected forum.

Original thread: [Announcement Multi Forum Select](https://community.mybb.com/thread-222166.html)

**Comment by effone** [comment]
Isn't it meant to be that way?
As much I can see the the assignment method of single announcement to multiple forums is not even in place. So simply making the control multi selectable is not sufficient.

**Comment by Ben-MyBB** [comment]
If it isn't possible to multi select announcement forums this can either be closed or marked as an Enhancement for 1.10.

**Comment by JordanMussi** [comment]
Yeah announcement is either for all forums (fid=0) or a single forum:

https://github.com/mybb/mybb/blob/36e8e7167c693ad0ac78dd05406e4cfa8fe0a9eb/install/resources/mysql_db_tables.php#L65-L67

Enhancement sounds good if it is popular enough :smile:

**Comment by arcalys** [comment]
Is this accepted as a feature then?

**Comment by Ben-MyBB** [comment]
I would suggest so if you are willing to work on it. I think announcements should be much more useful and function the exact same as a normal thread but elevated to the announcement area.On 2 Jul 2025, at 16:07, Axel Fontana ***@***.***> wrote:﻿arcalys left a comment (mybb/mybb#3599)
Is this accepted as a feature then?

**Comment by arcalys** [comment]
But wouldn't it make sense then to:

- Migrate the contents from the `announcements` table into the `threads` table
- Replace the `sticky` flag in the `threads` table with a `type` field, where the value could be:
  - `0`: normal thread
  - `1`: sticky
  - `2`: announcement
  - `3`: global announcement (visible in all forums)

*(Values are just for the sake of example.)*

**Comment by Ben-MyBB** [comment]
Yes it would but not sure what others thinkOn 2 Jul 2025, at 18:51, Axel Fontana ***@***.***> wrote:﻿arcalys left a comment (mybb/mybb#3599)
But wouldn't it make sense then to:

Migrate the contents from the announcements table into the threads table
Replace the sticky flag in the threads table with a type field, where the value could be:

0: normal thread
1: sticky
2: announcement
3: global announcement (visible in all forums)



(Values are just for the sake of example.)

**Comment by Sama34** [comment]
You have to consider that announcements are editable from other pages than a regular thread. There are also additional settings/permissions for announcements only.

But you have raised a good discussion topic: is the announcements feature even worth keeping ?

I would agree than maybe not much. Sticky threads already make the whole feature redundant.

**Comment by emregeldegul** [comment]
The "Announcements" feature is a very useful tool that simplifies the forum experience and makes it easier for administrators to establish order. It allows drawing attention to a specific section with highlighted content—without the need to pin, sticky, or lock threads. This reduces the moderation workload and offers a cleaner line of communication to users.

While its usage may have decreased, instead of removing it, improvements could be considered to make it more visible or functional.

In my opinion, it should not be removed.

**Comment by arcalys** [comment]
The thing is, from a database perspective, this doesn't make much sense. It feels like a thread and behaves like a thread, so it probably should be a thread.

That doesn't mean it can't be displayed the same way as the current announcements. There could still be a separate HTML element in the template for announcements, and another one for normal threads and stickies — even if they're all stored as threads in the database.

Either way works for me — I'm just thinking about simplifying the database schema.

**Comment by dvz** [comment]
They're usually treated separately (search results, recent threads, post relationships, etc.) - merging them may complicate the thread logic further.

Adding a pivot table would make it possible to select multiple forums.

However, they already support inheritance - would it be a widely used feature to warrant changes?

**Comment by Ben-MyBB** [comment]
I think they should function as normal threads but at the very least we should add the multi selection On 3 Jul 2025, at 13:02, dvz ***@***.***> wrote:﻿dvz left a comment (mybb/mybb#3599)
They're usually treated separately (search results, recent threads, post relationships, etc.) - merging them may complicate the thread logic further.
Adding a pivot table would make it possible to select multiple forums.
However, they already support inheritance - would it be a widely used feature to warrant changes?
